beanie/odm/documents.py

```diff
@@ -767,13 +766,13 @@ def update_all(
         )
 
     def set(
-        self: DocType,
+        self,
         expression: Dict[Union[ExpressionField, str, Any], Any],
         session: Optional[ClientSession] = None,
         bulk_writer: Optional[BulkWriter] = None,
         skip_sync: Optional[bool] = None,
         **kwargs,
-    ) -> Awaitable[DocType]:
+    ):
```

Comment:
Yes, you are right, the set() method returns a coroutine. Running under mypy with strict settings doesn't complain if we annotate the return to be _Coroutine[None, None, DocType]_.
Removing the type annotation here on this line effectively means typing it as Any, which would not be the best solution (effectively it's a workaround), since we know the return type here.
If you want, you may change it also for the inc() method since it's effectively the same.